### PR TITLE
[9.0][NEW] product_pricelist_direct_print: New module

### PR DESCRIPTION
**Pull request by carlosdauden**
Print price list from menu option, product templates, products or price lists.

@Tecnativa

### CODE REVIEW

File: product_pricelist_direct_print/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
+
+from . import wizard
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/__init__.py]: s/wizard/wizards

File: product_pricelist_direct_print/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "website": "http://www.tecnativa.com",
+    "author": "Tecnativa, "
+              "Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "depends": [
+        "product",
+    ],
+    "data": [
+        "views/report_product_pricelist.xml",
+        "wizard/product_pricelist_print_view.xml",
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/__openerp__.py]: s/wizard/wizards

File: product_pricelist_direct_print/README.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
+.. image:: https://img.shields.io/badge/licence-LGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/lgpl-3.0-standalone.html
+   :alt: License: LGPL-3
+
+=======================
+Product Pricelist Print
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/README.rst]: I think that this should be "Product Pricelist Direct Prient"

File: product_pricelist_direct_print/README.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+=======================
+
+Print price list from menu option, product templates, products or price lists
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/README.rst]: s/products/product variants

File: product_pricelist_direct_print/README.rst [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+* Go to *Sales > Configuration > Settings*
+* Scroll to *Quotations & Sales > Sale Price*
+* Set *Different prices per customer segment* or *Advanced pricing based on
+formula*.
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/README.rst]: add 2 spaces

File: product_pricelist_direct_print/README.rst [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+=======================
+
+Print price list from menu option, product templates, products or price lists
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+* Go to *Sales > Configuration > Settings*
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/README.rst]: add final dot

File: product_pricelist_direct_print/README.rst [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+* Scroll to *Quotations & Sales > Sale Price*
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/README.rst]: idem

File: product_pricelist_direct_print/README.rst [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+formula*.
+
+Usage
+=====
+
+To use this module, you have several options:
+
+#. Go to *Sales > Sales > Print Price List*
+#. Go to *Sales > Sales > Products*
+* Select products in list view
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/README.rst]: Add four blanck space

File: product_pricelist_direct_print/__openerp__.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Product Pricelist Print",
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/__openerp__.py]: s/Product Pricelist Print/Product Pricelist Direct Print

File: product_pricelist_direct_print/i18n/es.po [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+#: model:ir.model.fields,field_description:product_pricelist_direct_print.field_product_pricelist_print_product_ids
+#: model:ir.model.fields,field_description:product_pricelist_direct_print.field_product_pricelist_print_product_tmpl_ids
+#: model:ir.ui.view,arch_db:product_pricelist_direct_print.view_product_pricelist_print
+msgid "Products"
+msgstr "Productos"
+
+#. module: product_pricelist_direct_print
+#: model:ir.model.fields,field_description:product_pricelist_direct_print.field_product_pricelist_print_show_sale_price
+msgid "Show sale price"
+msgstr "Mostrar PVP"
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/i18n/es.po]: s/PVP/Precio de venta

File: product_pricelist_direct_print/views/report_product_pricelist.xml [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+<!-- Copyright 2017 Carlos Dauden <carlos.dauden@tecnativa.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+<template id="report_product_pricelist_document">
+    <t t-call="report.html_container">
+        <t t-call="report.internal_layout">
+            <div class="page">
+                <h2>Price List</h2>
+
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/views/report_product_pricelist.xml]: Remove this blank line

File: product_pricelist_direct_print/views/report_product_pricelist.xml [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+                                    <strong t-field="category.name"/>
+                                </td>
+                            </tr>
+                            <tr t-foreach="products and products.with_context(categ_id=category.id).filtered(lambda x: x.categ_id.id == x.env.context['categ_id']) or products.search([('categ_id','=',category.id)])"
+                                t-as="product">
+                                <td>
+                                    <t t-if="product.default_code">
+                                        [<span t-field="product.default_code"/>]
+                                    </t>
+                                    <span t-esc="product.name"/>
```
Comment (sergio-teruel) [inline product_pricelist_direct_print/views/report_product_pricelist.xml]: Use t-field for get translatable field

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: product_pricelist_direct_print/views/report_product_pricelist.xml (hunk 12 above)
Comment (pedrobaeza) [reply product_pricelist_direct_print/views/report_product_pricelist.xml]: Better to use `t-field='product.display_name'` to get product name with attribute values, or you will only get the template name.

File: product_pricelist_direct_print/README.rst (hunk 6 above)
Comment (carlosdauden) [reply product_pricelist_direct_print/README.rst]: I have removed all final dots

**Comment by carlosdauden** [comment]
@sergio-teruel  @pedrobaeza changes done

**Review by sergio-teruel — APPROVED** [review]
Thanks.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Empty categories shouldn't be shown in the report

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
One last weird detail is that if you don't put pricelist_id, list price is 0. We have several options:

* Print all pricelists if no pricelist is selected.
* Put the field as required.
* Hide the list price column and only print sale/cost price.

Other thing: please rename the checkbox to "Cost price", as it's the label in UI (although the technical name is standard_price)

**Comment by pedrobaeza** [comment]
Please check Travis

**Review by pedrobaeza — APPROVED** [review]

File: product_pricelist_direct_print/README.rst [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+
+Print price list from menu option, product templates, products variants or
+price lists
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+* Go to *Sales > Configuration > Settings*
```
Comment (rafaelbn) [inline product_pricelist_direct_print/README.rst]: Hi @carlosdauden , this is not important but just to say that the first is to install App Sales

**Review by rafaelbn — APPROVED** [review]
Very nice, tested in runbot 👍

File: product_pricelist_direct_print/README.rst (hunk 13 above)
Comment (carlosdauden) [reply product_pricelist_direct_print/README.rst]: Changes done

**Comment by bealdav** [comment]
Tested, works, units tests are ok, mergeable :+1: thanks

**Review by alexcbr2 — APPROVED** [review]
Functionallity tested OK

**Pull request merged**